moped-editor/src/views/projects/newProjectView/NewProjectMap.js

```diff
@@ -200,11 +276,57 @@ const NewProjectMap = ({
       : "default";
   };
 
+  /**
+   * Whenever a Save Action is initiated, save all drawn features
+   * */
+  useEffect(() => {
+    // Ignore empty states
+    if (!saveActionState) return;
+    // If the process has been already initiated, we don't need to go any further
+    if (saveActionState?.currentStep && saveActionState.currentStep > 1) return;
+    // It looks like this is the first step
+    if (
+      saveActionState?.initiateFeatureSave &&
+      saveActionState?.featuresSaved === false
+    ) {
+      saveDrawnPoints();
+    }
+  }, [saveActionState, saveDrawnPoints]);
+
   return (
     <Box className={noPadding ? classes.mapBoxNoPadding : classes.mapBox}>
-      {/* Render these controls outside ReactMapGL so mouse events don't propagate to the map */}
-      <div ref={mapControlContainerRef} className={classes.geocoderContainer} />
-      {renderLayerSelect()}
+      <style>
+        {!visibleLayerIds.includes("drawnByUser") &&
+          `g > circle { display: none; }`}
+        {!visibleLayerIds.includes("drawnByUserLine") &&
+          `g > path { display: none; }`}
+      </style>
```

Comment:
These are four lines that govern hiding/showing drawn points and lines in the user drop-down menu. Also not proud, I could have done this better but for now leaving this here for open criticism.

Reply:
The only part I don't love is the `<style>` tags. I wish there was a way we could use this same logic but use the [more common React](https://reactjs.org/docs/dom-elements.html#style) `style={inlineStyles}` prop. But I'm just not seeing how that would be possible in this situation. I think if you don't refactor, moving some of this Github comment into a code comment could be helpful.